discovery_engine_core/core/src/document.rs

```diff
@@ -63,23 +79,86 @@ pub struct Document {
     /// Stack from which the document has been taken.
     pub stack_id: StackId,
 
-    /// Position of the document from the source.
-    pub rank: usize,
+    /// Embedding from smbert.
+    pub smbert_embedding: Embedding1,
+
+    /// Resource this document refers to.
+    pub web_resource: WebResource,
+}
 
-    /// Text title of the document.
+/// Represents different kinds of resources like web, image, video, news,
+/// that are delivered by an external content API.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WebResource {
+    /// Title of the resource.
     pub title: String,
 
-    /// Text snippet of the document.
+    /// Snippet of the resource.
     pub snippet: String,
 
-    /// URL of the document.
-    pub url: String,
+    /// Url to reach the resource.
+    pub url: Url,
 
-    /// Domain of the document.
-    pub domain: String,
+    /// Url to display to the user.
+    // TODO: not sure it make sense to have this here, seems a ui decision
+    pub display_url: Url,
 
-    /// Embedding from smbert.
-    pub smbert_embedding: Embedding1,
+    /// Publishing date.
+    pub date_published: NaiveDateTime,
+
+    /// Provider.
+    pub provider: Option<WebResourceProvider>,
+
+    /// The rank of the domain of the source,
+    pub rank: usize,
+
+    /// How much the article match the query.
+    pub score: Option<f32>,
+
+    /// The country of the publisher.
+    pub country: String,
+
+    /// The language of the article.
+    pub language: String,
+
+    /// Main topic of the publisher.
+    pub topic: Topic,
+}
+
+impl TryFrom<Article> for WebResource {
+    type Error = Error;
+    fn try_from(article: Article) -> Result<Self, Self::Error> {
+        let media = article.media;
+
+        Ok(WebResource {
+            title: article.title,
+            snippet: article.excerpt,
+            date_published: article.published_date,
+            url: Url::parse(&article.link)?,
+            display_url: Url::parse(&article.clean_url)?,
+            rank: article.rank,
+            score: article.score,
+            country: article.country,
+            language: article.language,
+            topic: article.topic,
+            provider: Some(WebResourceProvider {
+                name: String::new(),
+                thumbnail: (!media.is_empty())
+                    .then(|| Url::parse(&media))
+                    .transpose()?,
+            }),
+        })
+    }
+}
+
+/// Represents the provider of a [`WebResource`].
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WebResourceProvider {
```

Comment:
It keeps the name of the provider and the link to the thumbnail of the logo of the provider. Is something we just copied from what was done in the app. Some change will be needed here because we don't have a name anymore with the current provider, but we need to sync with blue.

Reply:
If we get no official "name", then we can just parse the host name from the url.
We need to display the source for legal reasons, Bing used to provide this, cause they often hosted external providers via msn.com.

Ps the data structure may be altered, webResource and Provider are basically matching the old Bing api really.

Reply:
...if we do parse the host from the url, then it's also slightly tricky, for example:
washingtonpost.com
Or telegraph.co.uk
Or worse, media.telegraph.co.uk

That's hard to pour into a deterministic name of course, but imo it's fine to just display the full host name, whatever it is.

Reply:
We don't have the info from Article for this so I'm going to remove it.

Reply:
The api provide `clean_url`, that represent the url of the source (usually the base domain). We can export that as `source_url`.

Reply:
`source_url` is there!